### PR TITLE
Running mate-terminal on wayland

### PR DESCRIPTION
**Pull request by vladtcvs**
remove X11 dependence in id generation

remove signal window manager changed - it is incorrect for wayland case

### CODE REVIEW

**Comment by lukefromdc** [comment]
I can confirm that this works in both X11 and in Weston on my system. Using gtk3.19.5 at the moment.

**Comment by monsta** [comment]
> remove signal window manager changed - it is incorrect for wayland case

What happens on wayland if we leave it as is?

**Comment by raveit65** [comment]
Btw. m-t (gtk2) runs fine here in gnome-wayland session (fedora 23), so what are the benefits?

**Comment by lukefromdc** [comment]
The gtk3 version did NOT run in a wayland session without this fix in my testing as gtk3 actually uses wayland if a wayland session is running

**Comment by raveit65** [comment]
hmm, m-t (gtk3) w/o this patch runs well in gnome-wayland session when started from activities but not from a terminal.
```
[rave@f23-test ~]$ mate-terminal

** (mate-terminal:3049): WARNING **: Could not open X display
Segmentation fault (core dumped)
[rave@f23-test ~]$
```
With the patch m-t starts but the X warning is there which we can ignore.
```
[rave@f23-test ~]$ mate-terminal

** (mate-terminal:3241): WARNING **: Could not open X display
[rave@f23-test ~]$
```
Patched m-t runs fine in mate-session under X.
So, the PR looks fine for me.
@monsta ^^^^

**Comment by raveit65** [comment]
@vladtcvs 
Can you please answer to question from comment 3 ?

**Comment by vladtcvs** [comment]
May be you are using xwayland? Gtk2 can not use wayland nativly

**Comment by raveit65** [comment]
@vladtcvs 
> >    remove signal window manager changed - it is incorrect for wayland case

> What happens on wayland if we leave it as is?

??

**Comment by raveit65** [comment]
And yes, gnome-wayland session use X-Wayland

**Comment by vladtcvs** [comment]
@monsta 

> What happens on wayland if we leave it as is?

If we use wayland without x-wayland, we get crash of m-t

**Comment by vladtcvs** [comment]
Rebased on the new master branch

**Comment by monsta** [comment]
Personally, I'd use macros like ```GDK_IS_X11_SCREEN``` and ```GDK_IS_X11_WINDOW``` for checking if we're running on X11... I think it's better than just commenting the code.
The problem is that these macros aren't available in GTK+2... :confused:

**Comment by monsta** [comment]
However, if someone can confirm that **all** window managers out there support ```_NET_WM_STATE_FULLSCREEN```, then we can safely drop the code in question (and therefore make the ```View -> Fullscreen``` menu item always accessible).

**Comment by vladtcvs** [comment]
@monsta 

what about add --with-wm-support option and corresponding macros?

**Comment by monsta** [comment]
Ok, try https://github.com/mate-desktop/mate-terminal/pull/110 guys. Should cover all three cases (GTK+2, GTK+3, GTK+3 on Wayland).

**Comment by lukefromdc** [comment]
OK on X11 but got a segfault on a Weston session.

File: src/terminal.c [hunk 1]
```diff
@@ -606,11 +559,10 @@ main (int argc, char **argv)
...
 		/* Create a fake one containing a timestamp that we can use */
-		Time timestamp;
-
-		timestamp = slowly_and_stupidly_obtain_timestamp (GDK_DISPLAY_XDISPLAY (gdk_display_get_default ()));
-
-		options->startup_id = g_strdup_printf ("_TIME%lu", timestamp);
+		unsigned long timestamp = time(NULL);
+		unsigned rand_id = rand() & 0xFF;
+		options->startup_id = g_strdup_printf ("_TIME%lu",
+						       timestamp*16 + rand_id);
```
Comment (monsta) [inline src/terminal.c]: Hmm... is such algorithm needed? Multiplying timestamp by 16 and all that...

**Comment by raveit65** [comment]
@vladtcvs 
There are some open questions?

**Comment by monsta** [comment]
I don't see a point in keeping this open, at least for now.

**Comment by raveit65** [comment]
I got  a report in f25 about a crash in gnome session with wayland.
Here wayland is the default.
https://bugzilla.redhat.com/show_bug.cgi?id=1398584

**Comment by monsta** [comment]
We can open a report for tracking Wayland issues, but I think people will have to use XWayland as a workaround for some time.

**Comment by raveit65** [comment]
As far i know it is a x-wayland session with gnome in f25.

**Comment by monsta** [comment]
Don't they have Wayland by default there? Phoronix was overexcited about that recently.

**Pull request closed without merging**